mkosi/run.py

```diff
@@ -343,19 +299,49 @@ def bwrap(
             if ARG_DEBUG_SHELL.get():
                 run([*cmdline, "sh"], stdin=sys.stdin, check=False, env=env, log=False)
             raise e
-        finally:
-            # Clean up some stuff that might get written by package manager post install scripts.
-            if apivfs:
-                for f in ("var/lib/systemd/random-seed", "var/lib/systemd/credential.secret", "etc/machine-info"):
-                    # Using missing_ok=True still causes an OSError if the mount is read-only even if the
-                    # file doesn't exist so do an explicit exists() check first.
-                    if (apivfs / f).exists():
-                        (apivfs / f).unlink()
 
         return result
 
 
-def chroot_cmd(root: Path, *, options: Sequence[PathString] = (), network: bool = False) -> Sequence[PathString]:
+def apivfs_cmd(root: Path) -> list[PathString]:
+    cmdline: list[PathString] = [
+        "bwrap",
+        "--dev-bind", "/", "/",
+        "--chdir", Path.cwd(),
+        "--tmpfs", root / "run",
+        "--tmpfs", root / "tmp",
+        "--bind", os.getenv("TMPDIR", "/var/tmp"), root / "var/tmp",
+        "--proc", root / "proc",
+        "--dev", root / "dev",
+        "--ro-bind", "/sys", root / "sys",
+    ]
+
+    if (root / "etc/machine-id").exists():
+        # Make sure /etc/machine-id is not overwritten by any package manager post install scripts.
+        cmdline += ["--ro-bind", root / "etc/machine-id", root / "etc/machine-id"]
+
+    # If passwd or a related file exists in the apivfs directory, bind mount it over the host files while
+    # we run the command, to make sure that the command we run uses user/group information from the
+    # apivfs directory instead of from the host. If the file doesn't exist yet, mount over /dev/null
+    # instead.
+    for f in ("passwd", "group", "shadow", "gshadow"):
+        p = root / "etc" / f
+        if p.exists():
+            cmdline += ["--bind", p, f"/etc/{f}"]
+        else:
+            cmdline += ["--bind", "/dev/null", f"/etc/{f}"]
+
+    chmod = f"chmod 1777 {root / 'tmp'} {root / 'var/tmp'} {root / 'dev/shm'}"
+    # Make sure anything running in the root directory thinks it's in a container. $container can't always be
+    # accessed so we write /run/host/container-manager as well which is always accessible.
+    container = f"mkdir {root}/run/host && echo mkosi > {root}/run/host/container-manager"
```

Comment:
Nit
```suggestion
    container = f"mkdir {root}/run/host && echo mkosi >{root}/run/host/container-manager"
```
for the same style as the other shell stuff in systemd